typed/remove.go

```diff
@@ -20,61 +20,60 @@ import (
 )
 
 type removingWalker struct {
```

Comment:
Now that you have the interface, I think it'd make a lot of sense to have this actually be an implementation of the interface that takes a value and a fieldset, and then do the "removal" lazily by pretending things in the FS don't exist? (can be followup)

Reply:
Definitely part of the plan